admin.php

```diff
@@ -409,6 +409,7 @@ private function markRequested($request) {
      *
      */
     private function applyMatches() {
+        set_time_limit(120);
```

Comment:
I don't like messing with peoples php.ini without their explicit consent. Let's postpone this until we have settings, so the user can specify time limit themselves.

Reply:
It's not messing with the `php.ini`.
`php.ini` holds many settings which can be overridden within scripts. This is one example of overriding a setting only within the current script execution, just asking "give me 120 seconds more time to finish my business with this script" repeatedly.

Even PHP `ini_set()` does not change the actual `php.ini` file but only overrides settings for the current script execution. There is no API to change the actual `php.ini` file from within PHP. Except normal file functions if PHP has write access to that file.

Actually, from looking around a bit, it seems that this overriding of ini settings at execution time for the time of execution of one script from within that script is quite common practice.

The time limit here is not absolute. More time is requested at the beginning of "applyMatches" and the beginning of each call to "editFile", because the number of files, and with that the required execution time, can vary from a handful to many thousands.

Reply:
Sure I understand that you are not changing .ini itself. What I meant is that there might be some reason a parameter in the .ini is set to a certain value, which we are not aware of. So overriding it from our script feels a bit presumptuous.

Just to stress: I don't mind the feature, I just think DW admin should set this value. It shouldn't be hard-coded in the plugin.

Reply:
Okay, that makes sense.